addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/handler/DishwasherChannelSelector.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.miele.handler;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map.Entry;
+import java.util.TimeZone;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.miele.handler.MieleBridgeHandler.DeviceMetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+
+/**
+ * The {@link ApplianceChannelSelector} for dishwashers
+ *
+ * @author Karel Goderis - Initial contribution
+ */
+public enum DishwasherChannelSelector implements ApplianceChannelSelector {
+
+    PRODUCT_TYPE("productTypeId", "productType", StringType.class, true),
+    DEVICE_TYPE("mieleDeviceType", "deviceType", StringType.class, true),
+    STATE("state", "state", StringType.class, false),
+    PROGRAMID("programId", "program", StringType.class, false),
+    PROGRAMPHASE("phase", "phase", StringType.class, false),
+    START_TIME("startTime", "start", StringType.class, false),
+    DURATION("duration", "duration", DateTimeType.class, false) {
+        @Override
+        public State getState(String s, DeviceMetaData dmd) {
+            Date date = new Date();
+            SimpleDateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
```

Comment:
Maybe better to use DateTimeFormat from Joda Time here, which ist thread safe and can be cached, instead of creating every time a new date formatter?
http://stackoverflow.com/questions/6840803/simpledateformat-thread-safety